website/content/en/docs/golang/references/markers.md

```diff
@@ -8,11 +8,153 @@ This document describes [code markers][markers] supported by the SDK.
+
+	```go
+	type MemcachedSpec struct {
+		// Size is the size of the memcached deployment. <-- This will become Size's specDescriptors.description.
+		// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Number of pods",xDescriptors="urn:alm:descriptor:com.tectonic.ui:podCount,urn:alm:descriptor:io.kubernetes:custom"
+		Size int32 `json:"size"` // <-- Size's specDescriptors.path is inferred from this JSON tag.
+	}
+	```
+
+3. Let the SDK infer all unmarked paths on a field for a `customresourcedefinitions.specDescriptors` entry:
+
+	```go
+	type MemcachedSpec struct {
+		// Size is the size of the memcached deployment.
+		// +operator-sdk:csv:customresourcedefinitions:type=spec
+		Size int32 `json:"size"`
+	}
+	```
+
+	The SDK uses the `Size` fields' `json` tag name as `path`, `Size` as `displayName`, and field comments as `description`.
+
+4. A comprehensive example:
+	- Infer `path`, `description`, `displayName`, and `x-descriptors` for `specDescriptors` and `statusDescriptors` entries.
+	- Create three `resources` entries each with `kind`, `version`, and `name` values.
+
+	```go
+	// Represents a cluster of Memcached apps
+	// +operator-sdk:csv:customresourcedefinitions:displayName="Memcached App",resources={{Pod,v1,memcached-runner},{Deployment,v1,memcached-deployment}}
+	type Memcached struct {
+		metav1.TypeMeta   `json:",inline"`
+		metav1.ObjectMeta `json:"metadata,omitempty"`
+
+		Spec   MemcachedSpec   `json:"spec,omitempty"`
+		Status MemcachedStatus `json:"status,omitempty"`
+	}
+
+	type MemcachedSpec struct {
+		Pods MemcachedPods `json:"pods"`
+	}
+
+	type MemcachedStatus struct {
+		Pods 		 MemcachedPods `json:"podStatuses"`
+		// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Pod Count",xDescriptors="urn:alm:descriptor:com.tectonic.ui:podCount"
+		PodCount int 					 `json:"podCount"`
+	}
+
+	type MemcachedPods struct {
+		// Size is the size of the memcached deployment.
+		// +operator-sdk:csv:customresourcedefinitions:type=spec
+		// +operator-sdk:csv:customresourcedefinitions.type=status
+		Size int32 `json:"size"`
+	}
+	```
+
+	The generated `customresourcedefinitions` will look like:
+
+	```yaml
+	customresourcedefinitions:
+	  owned:
+	  - description: Represents a cluster of Memcached apps
+	    displayName: Memcached App
+	    kind: Memcached
+	    name: memcacheds.cache.example.com
+	    version: v1alpha1
+	    resources:
+	    - kind: Deployment
+	      name: memcached-deployment
+	      version: v1
+	    - kind: Pod
+	      name: memcached-runner
+	      version: v1
+	    specDescriptors:
+	    - description: The desired number of member Pods for the deployment.
+	      displayName: Size
+	      path: pods.size
+	    statusDescriptors:
+	    - description: The desired number of member Pods for the deployment.
+	      displayName: Size
+	      path: podStatuses.size
+	    - displayName: Size
```

Comment:
The `description` is a **_required_** property of both `specDescriptors` and `statusDescriptors`. We may need to add the default data for `description` field instead of omitting it.

Reply:
Just checked the [struct](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/vendor/github.com/operator-framework/api/pkg/operators/v1alpha1/clusterserviceversion_types.go#L90-L108) again and notice in fact only the `path` is the required property. So no concerns on the examples as well.

Reply:
I almost regret immediately after posting my previous comments. :(

Although "technically" only `path` is the required property, an empty `description` leads to an empty tooltip on the UI:

**Tooltip - with `description`:**
![1_tooltip___with_description](https://user-images.githubusercontent.com/5903705/88110057-6335f180-cb60-11ea-8f79-ab68dfeba5ea.png)

**Empty tooltip - without `description`:**
![2_empty_tooltip___without_description](https://user-images.githubusercontent.com/5903705/88110064-65984b80-cb60-11ea-805d-228dd818db88.png)

Can we at least include some simple text for `description` field in the example (if auto-generate "default" data is not an option)? Thanks!

Reply:
This is something I'd like to leave up to the [`OperatorHubValidator`](https://github.com/operator-framework/api/blob/master/pkg/validation/internal/operatorhub.go#L16), since it is perfectly valid to have a CSV without these descriptions.